sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -637,17 +637,24 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase {
   }
 
   def invalidFunctionArgumentsError(
-      name: String, expectedInfo: String, actualNumber: Int): Throwable = {
+      sqlExpr: Seq[Expression],
+      name: String,
+      expectedInfo: String,
+      actualNumber: Int): Throwable = {
     new AnalysisException(
-      errorClass = "_LEGACY_ERROR_TEMP_1042",
+      errorClass = "DATATYPE_MISMATCH.WRONG_NUM_ARGS",
       messageParameters = Map(
-        "name" -> name,
-        "expectedInfo" -> expectedInfo,
-        "actualNumber" -> actualNumber.toString))
+        "sqlExpr" -> sqlExpr.map(toPrettySQL(_)).mkString(","),
```

Comment:
Could you invoke `toSQLExpr()` instead of `toPrettySQL()`, please. Take a look at the other exceptions in the file.